### ISSUE TITLE
TestSource - Directory as source

### ISSUE THREAD

**Issue by zppinho** [kubeshop/testkube#4465]
**The Problem**
Currently, the only sources that are available for a Testkube Test are : "git", "file" or "string". In a scenario where I have a CI system that already clones the repository where the tests are , the requirement to clone the repository just to have the tests files introduces an unnecessary inefficiency to this system, furthermore, since the only way to clone a repository is by using a git-username and token this adds another thing that needs to be managed considering that the repository was cloned using ssh.

**The Solution**
A possible solution is to allow to create a TestSource based on a directory. This directory could be compressed and stored in the CR or somewhere else like the minio bucket, The Executors that utilize this Source would before running the tests uncompress the directory and run the tests.

**Comment by zppinho** [comment]
> thank you @zppinho we have a similar issue in backlog #4270 waiting for prioritization from @TheBrunoLopes and @jmorante-ks

In a perfect scenario I would prefer to set the Contents of the Source in the CR instead of minio

**Comment by zppinho** [comment]
Fair enough, but instead of limiting to minio you could allow any s3 bucket, with credentials and a path

**Comment by vsukhin** [comment]
minio and s3 are replacebale, you can work with minio or directly with s3, both are supported

**Comment by olensmar** [comment]
deprecated functionality..
